docker/complete/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM mesosphere/mesos:0.22.0-1.0.ubuntu1404
+MAINTAINER Mesosphere <support@mesosphere.io>
+
+RUN locale-gen en_US.UTF-8
+RUN dpkg-reconfigure locales
+ENV LANG en_US.UTF-8
+ENV LC_ALL en_US.UTF-8
+
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install --no-install-recommends -y \
+        g++ \
+        make \
+        curl \
+        mercurial \
+        git \
+        rsync \
+        patch \
+        wget \
+        && \
+    apt-get clean
+
+COPY . /kubernetes-mesos
+WORKDIR /kubernetes-mesos
+
+ENV GOROOT /usr/local/go
+ENV PATH $GOROOT/bin:$PATH
+
+RUN ./docker/complete/install-go.sh
```

Comment:
Docker uses pretty simple change detection while building. If the COPY above "changes something", the install scripts will be called again, including the download. It's much fast to put these two install lines above the local COPY.